### PR TITLE
Add searchForCustomers method

### PR DESCRIPTION
**Pull request by antoncazalet**
Implements [this](https://customer.io/docs/api/#operation/getPeopleFilter)

### CODE REVIEW

File: lib/types.ts [hunk 1]
```diff
@@ -16,22 +16,24 @@ export type SegmentFilter = {
 export type AttributeFilter = {
   field: string;
   operator: FilterOperator;
-  value?: string;
+  value?: string | number | boolean;
 };
 
-export type NotFilter = {
+export type NotSegment = {
```
[the remark this answers is not in the document]
Comment (antoncazalet) [reply lib/types.ts]: You're right ! Should be good now !

File: lib/types.ts [hunk 2]
```diff
@@ -16,22 +16,26 @@ export type SegmentFilter = {
...
 
 export type NotFilter = {
   segment?: SegmentFilter;
   attribute?: AttributeFilter;
 };
 
-export type FilterObject = SegmentFilter | AttributeFilter | NotFilter;
+export type FilterObject = SegmentFilter | AttributeFilter;
+
+export type NotFilterObject = {
```
Comment (mike-engel) [inline lib/types.ts:29]: This can replace `NotFilter` above since it's unused now. Otherwise this looks much better, thank you!

Comment (antoncazalet) [reply lib/types.ts:29]: I can't because it is still used in the docs, however It should probably be renamed "SegmentAndAttribute". What do you think ?

Comment (mike-engel) [reply lib/types.ts:29]: Which docs? If it's in the README in this repo, it can be removed/fixed

Comment (antoncazalet) [reply lib/types.ts:29]: The official Customer.io docs. On [this page](https://www.customer.io/docs/api/#operation/getPeopleFilter), in the NodeJS example section, you'll see a use case that I have put in the tests. And the _'old'_ **NotFilter** can be used in every context (and, or, not)

Comment (mike-engel) [reply lib/types.ts:29]: I actually think our documentation is wrong, and the structure if invalid. If you make a request with the example filters, the request will fail. I'm talking with the documentation team and will come back here once we figure it out. Thanks for your patience!

Comment (antoncazalet) [reply lib/types.ts:29]: Oh right ! Didn't tested the query, my bad ! Will update it, thank you :) !

**Comment by mike-engel** [comment]
Hi @antoncazalet, I wanted to check in on this to see if need any help 😄

**Comment by antoncazalet** [comment]
Hello ! I didn't get much time to update, I'll do it ASAP :)

**Pull request closed without merging**